### PR TITLE
Add linting to package

### LINKED ISSUE #37: Add linting to package
Linting and gh action

### PR DESCRIPTION
**Pull request by mzwiessele**

### CODE REVIEW

**Comment by neopostmodern** [comment]
do we really want to add linting to the master branch? i'd rather leave the old implementation as-is and only apply crucial fixes. thus i'd suggest rather merging this into the `refactor-to-django` branch.

**Review by neopostmodern — COMMENTED** [review]
- i'd have spared the `old_backend` from the linting but since it shouldn't really break anything i'm okay with applying it universally
- i don't have the competence to assess the `Makefile`, but it looks conceptually nice – maybe the README should refer to more `make` commands then?
- it weirds me out that `rest_framework` is being treated as a local package and placed *below* our actual local packages (probably due to alphabetical ordering) – is that really correct?

**Comment by mzwiessele** [comment]
Great catch! It wasn't pushing because of the precommit hooks...

**Comment by mzwiessele** [comment]
> * i'd have spared the `old_backend` from the linting but since it shouldn't really break anything i'm okay with applying it universally

Yeah it doesn't hurt :)

> * i don't have the competence to assess the `Makefile`, but it looks conceptually nice – maybe the README should refer to more `make` commands then?

Done

> * it weirds me out that `rest_framework` is being treated as a local package and placed _below_ our actual local packages (probably due to alphabetical ordering) – is that really correct?

Added into isort config, now it gets sorted with the third party packages.

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ uncontainerized local versions of the dependencies too.
 
 ### Virtual Environment
 
-Create and activate a virtual environment with Python 3.7+.
```
Comment (neopostmodern) [inline README.md]: Two trailing spaces in Markdown carry semantic meaning (simple line breaks) – could they be spared from linting? Maybe we exclude Markdown files from linting altogether? (Although removing the single trailing spaces in line 3/4 is good...)

Comment (mzwiessele) [reply README.md]: I think that is actually my editor, not the linting... Let me double check.

**Pull request merged**